dubbo-registry/dubbo-registry-nacos/src/main/java/org/apache/dubbo/registry/nacos/NacosRegistry.java

```diff
@@ -174,18 +182,33 @@ public List<URL> lookup(final URL url) {
     public void doRegister(URL url) {
         try {
             if (PROVIDER_SIDE.equals(url.getSide()) || getUrl().getParameter(REGISTER_CONSUMER_URL_KEY, false)) {
-                String serviceName = getServiceName(url);
                 Instance instance = createInstance(url);
+
+                Set<String> serviceNames = new HashSet<>();
+                // by default servicename is "org.apache.dubbo.xxService:1.0.0:"
+                String serviceName = getServiceName(url, false);
+                serviceNames.add(serviceName);
+
+                // in https://github.com/apache/dubbo/issues/14075
+                if (Boolean.parseBoolean(DUUBO_NACOS_EXPORT_NOTING_SUFFIX_SERVICENAME)) {
```

Comment:
Using `getUrl().getParameter("nacos.register-compatible", false)` would be better. That way, users can configure in a more standard way supported by Dubbo Config component:

```yaml
dubbo:
   registry:
      parameters:
         nacos.register-compatible: true
```

or

```shell
-Ddubbo.registry.parameters.nacos.register-compatible=true
```

or

```properties
dubbo.registry.parameters.nacos.register-compatible=true
```

Reply:
Please add some unit tests to verify this change. Thanks.